From 056aa56b6f2e52925cb2b278397bee4a593c5a89 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 24 Jan 2022 07:10:19 +0200
Subject: [PATCH] [SW-58071] gaudi3: get die id information

Gaudi3 has two die support and bootrom code will run on both die.
However the bootrom flow will differ based on which die its running.
In order to identify which die code is running, the patch add support
to read die id from register. Note that die id 0 will be always
master die.

Change-Id: I13f88f55b83a46273e9487c080b4737ed6675ba8
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index af41271..25d875c 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -21,7 +21,8 @@
 	.validation_pass = 0,
 	.efuse_read = 0,
 	.curr_img = 0,
-	.erom_boot_desc_flag = 0
+	.erom_boot_desc_flag = 0,
+	.die_id = 0
 };
 
 #define ARCBOOT_BRINGUP_ERROR		-1
@@ -116,7 +117,7 @@
 	security_disable_jtag_if();
 }
 
-static void arcboot_rom_init(void)
+__weak void arcboot_rom_init(void)
 {
 	uint32_t reg;
 
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 666b83f..c7bba15 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -38,3 +38,8 @@
 {
 	return 0;
 }
+
+void arcboot_rom_init(void)
+{
+	arcboot_rom_global.die_id = regmap_readl(mmD0_PARC_LOCAL_CFG_DIE_ID);
+}
diff --git a/include/arcboot.h b/include/arcboot.h
index a2e18e8..6f14e07 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -65,6 +65,7 @@
 	uint8_t efuse_read;
 	uint8_t curr_img;
 	uint8_t erom_boot_desc_flag;
+	uint8_t die_id;
 };
 
 enum arcboot_rom_bringup_proggress {
